### ISSUE TITLE
gojmx incompatible with apache/thrift v0.15

### ISSUE THREAD

**Issue by iangcarroll** [newrelic/nrjmx#94]
👋 I was trying out `gojmx` for the first time, and per the README ran `go get -u github.com/newrelic/nrjmx/gojmx`. However, it seems that the latest version of apache/thrift now requires a context to be passed to `iprot.ReadStructBegin` and similar calls, and `-u` forces the upgrade. As a result, it's unable to compile this package:

```
ian@ian jmx % go get -u github.com/newrelic/nrjmx/gojmx
go: downloading github.com/newrelic/nrjmx v2.0.0+incompatible
go: downloading github.com/newrelic/nrjmx/gojmx v0.0.0-20211221135603-df3fa035adca
go: downloading github.com/apache/thrift v0.13.0
go: downloading github.com/apache/thrift v0.15.0
# github.com/newrelic/nrjmx/gojmx/internal/nrprotocol
../go/pkg/mod/github.com/newrelic/nrjmx/gojmx@v0.0.0-20211221135603-df3fa035adca/internal/nrprotocol/nrjmx.go:182:37: not enough arguments in call to iprot.ReadStructBegin
	have ()
	want (context.Context)
../go/pkg/mod/github.com/newrelic/nrjmx/gojmx@v0.0.0-20211221135603-df3fa035adca/internal/nrprotocol/nrjmx.go:188:57: not enough arguments in call to iprot.ReadFieldBegin
	have ()
	want (context.Context)
../go/pkg/mod/github.com/newrelic/nrjmx/gojmx@v0.0.0-20211221135603-df3fa035adca/internal/nrprotocol/nrjmx.go:200:29: not enough arguments in call to iprot.Skip
	have (thrift.TType)
	want (context.Context, thrift.TType)
../go/pkg/mod/github.com/newrelic/nrjmx/gojmx@v0.0.0-20211221135603-df3fa035adca/internal/nrprotocol/nrjmx.go:339:33: not enough arguments in call to iprot.ReadFieldEnd
	have ()
	want (context.Context)
../go/pkg/mod/github.com/newrelic/nrjmx/gojmx@v0.0.0-20211221135603-df3fa035adca/internal/nrprotocol/nrjmx.go:343:32: not enough arguments in call to iprot.ReadStructEnd
	have ()
	want (context.Context)
../go/pkg/mod/github.com/newrelic/nrjmx/gojmx@v0.0.0-20211221135603-df3fa035adca/internal/nrprotocol/nrjmx.go:350:32: not enough arguments in call to iprot.ReadString
	have ()
	want (context.Context)
../go/pkg/mod/github.com/newrelic/nrjmx/gojmx@v0.0.0-20211221135603-df3fa035adca/internal/nrprotocol/nrjmx.go:359:32: not enough arguments in call to iprot.ReadString
	have ()
	want (context.Context)
../go/pkg/mod/github.com/newrelic/nrjmx/gojmx@v0.0.0-20211221135603-df3fa035adca/internal/nrprotocol/nrjmx.go:368:29: not enough arguments in call to iprot.ReadI32
	have ()
	want (context.Context)
../go/pkg/mod/github.com/newrelic/nrjmx/gojmx@v0.0.0-20211221135603-df3fa035adca/internal/nrprotocol/nrjmx.go:377:32: not enough arguments in call to iprot.ReadString
	have ()
	want (context.Context)
../go/pkg/mod/github.com/newrelic/nrjmx/gojmx@v0.0.0-20211221135603-df3fa035adca/internal/nrprotocol/nrjmx.go:386:32: not enough arguments in call to iprot.ReadString
	have ()
	want (context.Context)
../go/pkg/mod/github.com/newrelic/nrjmx/gojmx@v0.0.0-20211221135603-df3fa035adca/internal/nrprotocol/nrjmx.go:386:32: too many errors
```

Not updating the dependency (i.e. running only `go get github.com/newrelic/nrjmx/gojmx`) seems to temporarily fix it, but just wanted to flag it. Happy holidays!

**Comment by josemore** [comment]
thanks for reporting this @iangcarroll, we're still working on the new nrjmx version based on apache/thrift and we'll have a look at this / cc @cristianciutea

**Comment by cristianciutea** [comment]
fixed by #104
